packages/taiko-client/proposer/proposer.go

```diff
@@ -458,12 +453,23 @@ func (p *Proposer) ProposeTxListOntake(
 		return errors.New("insufficient prover balance")
 	}
 
-	txCandidate, err := p.txBuilder.BuildOntake(ctx, txListsBytesArray)
+	txCandidate, cost, err := p.buildCheaperOnTakeTransaction(ctx, txListsBytesArray)
```

Comment:
I saw it was always get cheaper txCandidate from `buildCheaperOnTakeTransaction`.
How about making this feature optional?